File: demo-workspaces/ai_cart_project/src/behavior_demo.py
def calculate_total(price, quantity):
    """Calculates total price for items."""
    return price * quantity

def is_eligible_for_discount(subtotal, is_vip):
    """Determines discount eligibility."""
    if is_vip:
        return True
    return subtotal >= 100.0

def parse_item_count(value):
    """Parses item count and validates non-negative value."""
    val = int(value)
    if val < 0:
        raise ValueError("Item count cannot be negative")
    return val

def summarize_cart(items):
    """Summarizes item quantities."""
    if not isinstance(items, list):
        raise TypeError("items must be a list")
    return {"total_items": len(items), "items": items}

File: demo-workspaces/ai_cart_project/src/behavior_demo.ts
export function calculateTotal(price: number, quantity: number): number {
  return price * quantity;
}

export const isEligibleForDiscount = (subtotal: number, isVip: boolean): boolean => {
  if (isVip) return true;
  return subtotal >= 100.0;
};

export function parseItemCount(value: any): number {
  const count = parseInt(value, 10);
  if (isNaN(count)) {
    throw new TypeError("Item count must be a valid number");
  }
  if (count < 0) {
    throw new RangeError("Item count cannot be negative");
  }
  return count;
}

export const summarizeCart = (items: any): { total_items: number; items: any[] } => {
  if (!Array.isArray(items)) {
    throw new TypeError("items must be an array");
  }
  return { total_items: items.length, items };
};
